### PR TITLE
implement bake-kdf function

### PR DESCRIPTION
**Pull request by makavity**
Need https://github.com/RustCrypto/hashes/pull/520 for merge

### CODE REVIEW

File: bake-kdf/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+pub fn bake_kdf(x: &[u8], s: &[u8], c: u128) -> Result<[u32; 8], Error> {
+    let mut hasher = BeltHash::default();
+    hasher.update(x);
+    hasher.update(s);
+    let y = hasher.finalize_fixed();
+
+    let d: [u32; 3] = [0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF];
+
+    let mut c = to_u32::<4>(&c.to_be_bytes());
+    c.reverse();
```
Comment (newpavlov) [inline bake-kdf/src/lib.rs:161]: Are you sure the reversal is needed on BE targets? Ideally, we should test correctness on BE targets using either MIRI or cross.

Comment (makavity) [reply bake-kdf/src/lib.rs:161]: I will check tomorrow, thank you. Not sure in it

Comment (makavity) [reply bake-kdf/src/lib.rs:178]: @newpavlov fiex.

**Comment by makavity** [comment]
@newpavlov hey, could you take a look?

**Comment by makavity** [comment]
@newpavlov hey, could you check it? Need it for PAKE algorithm.

**Comment by makavity** [comment]
Thank you!

**Pull request merged**